libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -253,7 +259,8 @@ void ImmutableValidator::analyseVariableReference(VariableDeclaration const& _va
 
 void ImmutableValidator::checkAllVariablesInitialized(solidity::langutil::SourceLocation const& _location)
 {
-	for (ContractDefinition const* contract: m_currentContract.annotation().linearizedBaseContracts)
+	for (ContractDefinition const* contract: m_mostDerivedContract.annotation().linearizedBaseContracts | ranges::views::reverse)
+	{
```

Comment:
But now here: why do we look at all the contracts in the hierarchy? What would we miss if we only looked at the current/most derived contract alone?

Each contract in the hierarchy will be visited as the "most derived" one eventually, won't it?
And each one has to initialize exactly those immutables in the contract that it is defined it, no more, no less, right?
So I guess we could just visit only that contract relative to those state variables here each time?

Reply:
Or put differently:
```
contract C {
        uint256 immutable x;
        constructor() {
        }
}
contract D is C {
}
```
currently produces two errors, one relative to ``C``, one relative to ``D``. The one relative to ``D`` is redundant: the issue can only be fixed in ``C`` anyways and ``D`` cannot be compiled without ``C`` anyways.